File: ELEC881_Copilot/src/Function Name Masked/Python/q1-two-sum.py
class Solution(object):
    def test(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        for i, num in enumerate(nums):
            if target - num in d:
                return [d[target - num], i]
            d[num] = i
        return []
    
    def twoSum(self, nums, target):
        return self.test(nums, target)

File: ELEC881_Copilot/src/Function Name Masked/Java script (JS)/q1-two-sum.js
var test = function(nums, target) {
    var map = {};
    for (var i = 0; i < nums.length; i++) {
        var diff = target - nums[i];
        if (map[diff] !== undefined) {
            return [map[diff], i];
        }
        map[nums[i]] = i;
    }
    return [];
    
};

module.exports = test;

